File: java/machinelearning/machinelearning_impl/src/main/java/com/scathon/ml/java/cluster/kmeans/KmeansConfiguration.java
package com.scathon.ml.java.cluster.kmeans;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class KmeansConfiguration {
    public static String FILE_PATH;
    public static int MAX_ITERATION_NUM;
    public static int POINT_DIMENSION;
    public static int CLUSTER_NUM;

    static {
        try (InputStream configInputStream = KmeansConfiguration.class.getClassLoader().getResourceAsStream("cluster" +
                "/kmeans" +
                "/config/kmeans.properties");) {
            Properties prop = new Properties();
            prop.load(configInputStream);
            FILE_PATH = prop.getProperty("kmeans.training.file.path");
            MAX_ITERATION_NUM = Integer.parseInt(prop.getProperty("kmeans.max.iteration.num"));
            POINT_DIMENSION = Integer.parseInt(prop.getProperty("kmeans.point.dimension"));
            CLUSTER_NUM = Integer.parseInt(prop.getProperty("kmeans.cluster.num"));
        } catch (IOException e) {
            System.err.println("加载kmeans算法配置文件失败！");
            System.exit(-1);
        }
    }
}

File: java/machinelearning/machinelearning_impl/src/main/scala/com/scathon/ml/scala/cluster/kmeans/config/KmeansConfiguration.scala
package com.scathon.ml.scala.cluster.kmeans.config

import java.util.Properties

import org.apache.commons.lang3.StringUtils


object KmeansConfiguration {
   var dimension: Int = 0
   var clusterNum: Int = 0
   var filePath: String = StringUtils.EMPTY
   var maxIterationNum: Int = 0
   var configFilePath: String = "cluster/kmeans/config/kmeans.properties"
   val prop = new Properties()
   prop.load(KmeansConfiguration.getClass.getClassLoader.getResourceAsStream(configFilePath))
   dimension = prop.getProperty("kmeans.point.dimension").toInt
   clusterNum = prop.getProperty("kmeans.cluster.num").toInt
   filePath = prop.getProperty("kmeans.training.file.path")
   maxIterationNum = prop.getProperty("kmeans.max.iteration.num").toInt
}
